mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala

```diff
@@ -80,6 +83,17 @@ private object DTO {
       mailboxId,
       expunged.mapValues(_.toJava).asJava)
   }
+
+  case class FlagsUpdated(sessionId: SessionId, user: User, path: MailboxPath, mailboxId: MailboxId, uids: List[MessageUid],
+                          updatedFlags: List[DTOs.UpdatedFlags]) extends Event {
+    override def toJava: JavaEvent = new JavaFlagsUpdated(
+      sessionId,
+      user,
+      path.toJava,
+      mailboxId,
+      uids.asJava,
```

Comment:
We already have the list of UIDs as part of the updated flags list.

Does it makes sens to send the uid list two times in the JSON?

I do think that 

```
    private static final String DEFAULT_EVENT_JSON =
        "{" +
        "  \"FlagsUpdated\": {" +
        "    \"path\": {" +
        "      \"namespace\": \"#private\"," +
        "      \"user\": \"user\"," +
        "      \"name\": \"mailboxName\"" +
        "    }," +
        "    \"mailboxId\": \"18\"," +
//        "    \"uids\": [123456, 654321]," + // can be removed as the information is in updatedFlags
        "    \"sessionId\": 42," +
        "    \"updatedFlags\": [" +
        "      {" +
        "        \"uid\": 123456," +
        "        \"modSeq\": 35," +
        "        \"oldFlags\": [\"Old Flag 1\", \"\\\\Deleted\", \"\\\\Seen\"]," +
        "        \"newFlags\": [\"New Flag 1\", \"\\\\Answered\", \"\\\\Draft\"]" +
        "      }," +
        "      {" +
        "        \"uid\": 654321," +
        "        \"modSeq\": 36," +
        "        \"oldFlags\": [\"Old Flag 2\", \"\\\\Flagged\", \"\\\\Recent\"]," +
        "        \"newFlags\": [\"New Flag 2\", \"\\\\Answered\", \"\\\\Seen\" ]" +
        "      }" +
        "    ]," +
        "    \"user\": \"user\"" +
        "  }" +
        "}";
```

Reply:
If the content of `uids` and list of uids in `UpdatedFlags` are the same, Then is it legal to refactor `EventFactory` ? I see factory method carries both list of uids and list of updatedFlags.

Reply:
Maybe `FlagsUpdatedEvent` should just tale a List<UpdatedFlags> and compute `List<MessageUid>` from it, don't you think?

Reply:
I will do it in separated PR

Reply:
#2055